packages/story-editor/src/components/panels/design/layer/layer.js

```diff
@@ -420,11 +430,19 @@ function Layer({ element }) {
       {isRenameable && isLayerNamingEnabled ? (
         <LayerInputWrapper>
           <LayerIconWrapper>
-            <LayerIcon
-              element={element}
-              getProxiedUrl={getProxiedUrl}
-              currentPageBackgroundColor={currentPageBackgroundColor}
-            />
+            <div
+              style={{
+                ...(hasMask
+                  ? { position: 'relative', clipPath: `url(#${maskId})` }
+                  : {}),
+              }}
+            >
```

Comment:
I'd wrapper that we did this with a styled-component element, and then optionally added that in an extra component:

```js
const MaskedIconWrapper = styled.div`
  position: relative;
  clipPath: url(#${({ maskId }) => maskId});
`;
...
function IconWrapper({ hasMask, maskId, children }) {
  if (!hasMask) {
    return children;
  }
  return (
    <MaskedIconWrapper maskId={maskId}>
      {children}
    </MaskedIconWrapper>
  );
}
...
  <LayerIconWrapper>
    <IconWrapper hasMask={hasMask} maskId={maskId}>
      <LayerIcon ... />
    </IconWrapper>
  </LayerIconWrapper>
```

It just seems a bit cleaner to me. You could potentially even move resolution of `hasMask` and `maskId` into this component just pass `element` to it.